### PR TITLE
Add type hint for result object in lattice.py

### PR DESCRIPTION
**Pull request by Udayan853**
Add type hint for result_object in covalent/_workflow/lattice.py inside the _server_dispatch method.

Issue: https://github.com/AgnostiqHQ/covalent/issues/60

### CODE REVIEW

File: covalent/_workflow/lattice.py [hunk 1]
```diff
@@ -343,7 +343,7 @@ def dispatch_sync(self, *args, **kwargs) -> "Result":
 
         return get_result(self.dispatch(**kwargs), self.metadata["results_dir"], wait=True)
 
-    def _server_dispatch(self, result_object) -> str:
+    def _server_dispatch(self, result_object:Result) -> str:
```
Comment (FyzHsn) [inline covalent/_workflow/lattice.py:346]: There should be a space between the parameter and the parameter type. Can you run the tests and ensure that there are no circular import issues? If there are, you might have to use the string representation of result as in line 324.

**Review by FyzHsn — CHANGES_REQUESTED** [review]

File: covalent/_workflow/lattice.py (hunk 1 above)
Comment (Udayan853) [reply covalent/_workflow/lattice.py]: Fixed the issues with circular import by using string representation of result, as you had suggested. I also ran the tests to ensure there were no problems regarding the same.

**Review by FyzHsn — APPROVED** [review]
LGTM!

**Review by FyzHsn — COMMENTED** [review]
You just need to update the Changelog and Version files and this is good to be merged. This would count as an increment in the patch version.

**Comment by Udayan853** [comment]
Done!

**Review by FyzHsn — APPROVED** [review]

**Pull request merged**